app/_src/gateway/key-concepts/routes/index.md

```diff
@@ -15,6 +15,96 @@ For example, if you have an external application and an internal application tha
 
 In the example above, two routes can be created, say `/external` and `/internal`, and both routes can point to `example_service`. A policy can be configured to limit how often the `/external` route is used and the route can be communicated to the external client for use. When the external client tries to access the service via {{site.base_gateway}} using `/external`, they are rate limited. But when the internal client accesses the service using {{site.base_gateway}} using `/internal`, the internal client will not be limited.
 
+## How {{site.base_gateway}} routes requests in `traditional_compat` mode
+
+For each incoming request, {{site.base_gateway}} must determine which
+service gets to handle it based on the routes that are defined. The
+`traditional_compat` mode routing process is described in the
+following sections.
+
+The actual router implementation optimizes the process and caches
+results to avoid having to scan all routes repeatedly. When using the
+`expressions` router, route priorities are explicitly defined and this
+description doesn't apply. When you use the 'expressions' router,
+routes are converted from the `traditional_compat` format to the
+`expressions` format.
```

Comment:
This last sentence is a bit confusing now.  The `expressions` router is always used, but the mode in which the configuration is specified can be changed between `expressions` and `traditional_compat`.  This PR only describes how the `expressions` router operates in `traditional_compat` mode.  Underneath, the routing specifications in are converted to the native `expressions` format.

Maybe we need to move this into a completely new section that describes the two routers ('traditional' and 'expressions'), the three modes ('traditional', 'traditional_compat', and 'expressions') and how they correlate.  With that out of the way, this description of `traditional_compat` mode will probably be less confusing.

@cloudjumpercat @kikito thoughts?

Reply:
@hanshuebner I think I originally edited it to remove the word "underneath", so maybe that's why it's confusing now? If we wanted to add a sentence like "Underneath, the routing specifications in are converted to the native expressions format." back in, I'd try to use something like "At the ___ level," or something like that instead of "underneath".

> Maybe we need to move this into a completely new section that describes the two routers ('traditional' and 'expressions'), the three modes ('traditional', 'traditional_compat', and 'expressions') and how they correlate. With that out of the way, this description of traditional_compat mode will probably be less confusing.

I think if we wanted to do that, I almost think this content could be moved to a different page. I think the original intent of this "Routes Overview" page was just a general overview of routes and 'traditional', 'traditional_compat', and 'expressions' seem a lot more specific.